include/dem/disable_particle_contact.h

```diff
@@ -0,0 +1,280 @@
+   * floating walls or mesh
+   */
+  enum mobility_status : unsigned int
+  {
+    inactive = 0, // used for cells and nodes
+    active   = 1, // used for cells and nodes
+    mobile   = 2, // used for cells and nodes
+    empty    = 3  // used for nodes only
+  };
+
+  /**
+   * @brief Create or update a set of the active and ghost cells so that we don't
+   * have to loop over all the cells in the triangulation for the granular
+   * temperature and solid fraction calculation, and during the identification
+   * of the mobility status. This set prevent 4 iteration steps over all the
+   * cells  + the verification if the cell is locally owned, ghost or not.
+   * This set is updated at every load balance step since cells are
+   * redistributed among processors.
+   *
+   * @param background_dh The DoFHandler of the background grid
+   */
+  void
+  update_active_ghost_cell_set(const DoFHandler<dim> &background_dh);
+
+  /**
+   * @brief Carries out the identification of the mobility status of each cell
+   * through a node-based identification and check. Only the active and ghost
+   * cells are processed and only mobile and active cells are stored in the map
+   * since the cells that are not stored in the map are considered as empty or
+   * inactive.
+   *
+   * The following 4 checks (search loops) are done:
+   *
+   * 1. Check if the cell is empty (n_particle = 0), if so, nodes and cells are
+   * flagged as empty mobility status (3) (empty cells are not stored in the
+   * map).
+   *
+   * 2. Check if the cell is mobile by criteria (average granular temperature >
+   * threshold, solid fraction < threshold or has at least one empty node from
+   * previous check), if so, nodes are flagged and cells are stored with mobile
+   * mobility status (2)
+   *
+   * 3. Check if the cell is mobile by neighbor (at least a node is flagged as
+   * mobile from previous check), if so, cells are stored in map as mobile
+   * status (2) and nodes that are not mobile are flagged as active (1)
+   *
+   * 4. Check if the cell is active (at least a node is flagged as active from
+   * previous check), if so, cells are stored with active status in the map (1)
+   *
+   * The remaining cells are inactive (0), there are not stored in the map.
+   *
+   * @param background_dh The dof handler of the background grid
+   *
+   * @param particle_handler The particle handler that contains all the particles
+   *
+   * @param mpi_communicator The MPI communicator
+   */
+  void
+  identify_mobility_status(
+    const DoFHandler<dim> &                background_dh,
+    const Particles::ParticleHandler<dim> &particle_handler,
+    const unsigned int                     n_active_cells,
+    MPI_Comm                               mpi_communicator);
+
+  /**
+   * @brief Find the mobility status of a cell
+   *
+   * @param cell The iterator of the cell that needs mobility evaluation
+   */
+  inline unsigned int
+  check_cell_mobility(
```

Comment:
With the new unordered_map, you will no longer need this. Just use the key->value mechanism.